src/main/java/io/kaitai/struct/IKaitaiEnum.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2015-2024 Kaitai Project: MIT license
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package io.kaitai.struct;
+
+/**
+ * Interface, implemented by every enum class generated by Kaitai Struct Compiler,
+ * which provides assess to the underlying integer value.
+ *
+ * Each generated enum is represented by it's own interface and pair of nested
+ * classes: enum class and ordinal class:
+ *
+ * <code><pre>
+ * // marker interface
+ * public interface <kaitai-enum-name> extends IKaitaiEnum {
+ *     // marker class for values not explicitly listed in KSY
+ *     public final class Unknown extends IKaitaiEnum.Unknown implements <kaitai-enum-name> {
+ *         Unknown(long id) { super(id); }
+ *     }
+ *
+ *     public enum Known implements <kaitai-enum-name> {
+ *         // ...values...
+ *
+ *         Known(long id) { super(id); }
+ *
+ *         @Override
+ *         public long id() { ... }
+ *     }
+ *     // returns either Known.<...> or Unknown(id)
+ *     // Each Unknown created only once
+ *     public static <kaitai-enum-name> byId(long id) {
+ *         // ...
+ *     }
+ * }
+ * </pre></code>
+ *
+ * @since 0.11
+ */
+public interface IKaitaiEnum {
+    /**
+     * Base class for all unknown variants of Kaitai-generated enums.
+     *
+     * If you want to determine is the variant that you have is known or not,
+     * you can use {@code <variant> instanceof IKaitaiEnum.Unknown}. The {@link #name()}
+     * of such variant is always {@code null}.
+     */
+    public abstract class Unknown implements IKaitaiEnum {
+        protected final long id;
+        protected Unknown(long id) { this.id = id; }
+
+        @Override
+        public String name() { return null; }
```

Comment:
`toString` is implemented by generator as `<kaitai-java-enum-name>(<enum-long-value>)`.
`equals` now also implemented by generator (because we do not want that different classes be equal).

`hashCode` now implemented here as `Long.hashCode(this.id)`.